Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp

```diff
@@ -1442,8 +1442,14 @@ void NetworkProcessProxy::removeSession(WebsiteDataStore& websiteDataStore)
 {
     m_websiteDataStores.remove(websiteDataStore);
 
-    if (canSendMessage())
-        send(Messages::NetworkProcess::DestroySession { websiteDataStore.sessionID() }, 0);
+    if (canSendMessage()) {
+        if (auto completionHandler = websiteDataStore.takeCompletionHandlerForRemovalFromNetworkProcess()) {
```

Comment:
nit: I think this may look a bit better if we avoided the duplication:
```
sendWithAsyncReply(Messages::NetworkProcess::DestroySession { websiteDataStore.sessionID() }, [completionHandler = websiteDataStore.takeCompletionHandlerForRemovalFromNetworkProcess()]() mutable {
    if (completionHandler)
        return completionHandler({ });
});
```